build/loaders/worker-loader.js

```diff
@@ -0,0 +1,136 @@
+/** Copyright (c) 2018 Uber Technologies, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+/* eslint-env node */
+
+const path = require('path');
+const loaderUtils = require('loader-utils');
+
+const NodeTargetPlugin = require('webpack/lib/node/NodeTargetPlugin');
+const SingleEntryPlugin = require('webpack/lib/SingleEntryPlugin');
+const WebWorkerTemplatePlugin = require('webpack/lib/webworker/WebWorkerTemplatePlugin');
+
+class WorkerLoaderError extends Error {
+  constructor(err) {
+    super(err);
+
+    this.name = err.name || 'Loader Error';
+    this.message = `${err.name}\n\n${err.message}\n`;
+    /* $FlowFixMe */
+    this.stack = false;
+  }
+}
+
+const getWorker = (file, content, options) => {
+  const publicPath = options.publicPath
+    ? JSON.stringify(options.publicPath)
+    : '__webpack_public_path__';
+
+  const publicWorkerPath = `${publicPath} + ${JSON.stringify(file)}`;
+
+  if (options.inline) {
+    const InlineWorkerPath = JSON.stringify(
+      `!!${path.join(__dirname, 'InlineWorker.js')}`
+    );
+
+    const fallbackWorkerPath =
+      options.fallback === false ? 'null' : publicWorkerPath;
+
+    return `require(${InlineWorkerPath})(${JSON.stringify(
+      content
+    )}, ${fallbackWorkerPath})`;
+  }
+  return `${publicWorkerPath}`;
+};
+
+module.exports = function loader() {};
+
+module.exports.pitch = function(request /* : any*/) {
+  const options = loaderUtils.getOptions(this) || {};
+
+  if (!this.webpack) {
+    throw new WorkerLoaderError({
+      name: 'Worker Loader',
+      message: 'This loader is only usable with webpack',
+    });
+  }
+
+  this.cacheable(false);
+
+  const cb = this.async();
+
+  const filename = loaderUtils.interpolateName(
+    this,
+    options.name || '[hash].worker.js',
+    {
+      context: options.context || this.rootContext || this.options.context,
+      regExp: options.regExp,
+    }
+  );
+
+  const worker = {};
+
+  worker.options = {
+    filename,
+    chunkFilename: `[id].${filename}`,
+    namedChunkFilename: null,
```

Comment:
I think if you add `globalObject: "self"` here, this will yield the result you want